src/torrent_handle.cpp

```diff
@@ -363,7 +363,7 @@ namespace libtorrent
 
 	void torrent_handle::queue_position_set(int p) const
 	{
-		TORRENT_ASYNC_CALL1(set_queue_position, p < 0 ? 0 : p);
+		TORRENT_ASYNC_CALL1(set_queue_position, p);
```

Comment:
I see, you moved this check into the libtorrent side. I'm not convinced re-interpreting negative numbers as 0 is reasonable. I think having an assert (TORRENT_ASSERT_PRECOND) and just ignore the call when made with an invalid argument, might make more sense